shared-module/_pixelbuf/PixelBuf.c

```diff
@@ -218,17 +230,36 @@ void _pixelbuf_set_pixel(pixelbuf_pixelbuf_obj_t* self, size_t index, mp_obj_t v
     _pixelbuf_set_pixel_color(self, index, r, g, b, w);
 }
 
-void common_hal__pixelbuf_pixelbuf_set_pixels(mp_obj_t self_in, size_t start, mp_int_t step, size_t slice_len, mp_obj_t* values) {
+void common_hal__pixelbuf_pixelbuf_set_pixels(mp_obj_t self_in, size_t start, mp_int_t step, size_t slice_len, mp_obj_t* values, bool flattened) {
     pixelbuf_pixelbuf_obj_t* self = native_pixelbuf(self_in);
-    for (size_t i = 0; i < slice_len; i++) {
-        _pixelbuf_set_pixel(self, start, values[i]);
-        start+=step;
+    mp_obj_iter_buf_t iter_buf;
+    mp_obj_t iterable = mp_getiter(values, &iter_buf);
+    mp_obj_t item;
+    size_t i = 0;
+    mp_obj_tuple_t *tuple;
+    uint bpp = self->bytes_per_pixel;
+    if (flattened) {
+        tuple = MP_OBJ_TO_PTR(mp_obj_new_tuple(bpp, NULL));
```

Comment:
This is a memory allocation. Can you pass in a stack allocated tuple instead?

Reply:
By allocating a static tuple in pixelbuf_pixelbuf_subscr, perhaps?  The maximum length would be 4 items.

Reply:
Passing a stack allocated tuple would be a problem if the callee saved it, since it would become immediately

Reply:
And a statically allocated tuple would disobey mutability when given new content

Reply:
The list/tuple here is only used during `common_hal__pixelbuf_pixelbuf_set_pixels`.  However, as it was part of an attempt to avoid having to refactor `_pixelbuf_parse_color` and `_pixelbuf_set_pixel_color`, i think i'll just go refactor _pixelbuf_parse_color.

Reply:
@jepler My understanding is that it's short lived so stack allocated would be ok.

Reply:
OK, my objection is irrelevant -- I assumed the value was either escaping as a return value or passed to python code as a parameter.